src/main/java/org/opensearch/neuralsearch/search/HybridTopScoreDocCollector.java

```diff
@@ -115,6 +117,9 @@ public void collect(int doc) throws IOException {
                     collectedHitsPerSubQuery[i]++;
                     PriorityQueue<ScoreDoc> pq = compoundScores[i];
                     ScoreDoc currentDoc = new ScoreDoc(doc + docBase, score);
+                    if (score > maxScore) {
```

Comment:
nit: ` maxScore = Math.max(score, maxScore)` for readability

Reply:
Sure I will add the change.